### PR TITLE
6 - PATCH /api/articles/:article_id implementation

### PR DESCRIPTION
**Pull request by Daniel-N98**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -96,3 +96,78 @@ describe("GET /api/users", () => {
...
+          created_at: "2020-10-16T05:03:00.000Z",
+          votes: 10,
+        };
+        expect(body.updated_article).toEqual({
```
Comment (popatre) [inline __tests__/app.test.js]: Key wise, it needs to be `article`

File: __tests__/app.test.js [hunk 2]
```diff
@@ -96,3 +96,78 @@ describe("GET /api/users", () => {
...
+    const newVotes = { inc_votes: true };
+    return request(app)
+      .patch("/api/articles/NotAnID")
+      .send(newVotes)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.message).toBe("Bad request");
+      });
+  });
+});
```
Comment (popatre) [inline __tests__/app.test.js:183]: Just needs one more test:

- [x] 200/400 inc_vote key missing - either respond with an error or make no change.

File: controllers/app.controller.js [hunk 3]
```diff
@@ -34,3 +35,13 @@ exports.fetchUsers = (request, response, next) => {
     })
     .catch((error) => next(error));
 };
+
+exports.patchArticle = (request, response, next) => {
+  const { article_id } = request.params;
+  const { inc_votes } = request.body;
+  patchArticleById(article_id, inc_votes)
+    .then((updated_article) => {
+      response.status(200).send({ updated_article });
```
Comment (popatre) [inline controllers/app.controller.js]: key needs to be `article`

File: models/app.model.js [hunk 4]
```diff
@@ -18,3 +18,25 @@ exports.fetchArticleById = (article_id) => {
...
+  if (currentVotes.rowCount === 0) {
+    return Promise.reject({ code: 404, message: "Article not found" });
+  }
+  currentVotes = currentVotes.rows[0].votes;
+  return db
+    .query(
+      "UPDATE articles SET votes = $1 WHERE article_id = $2 RETURNING *;",
+      [currentVotes + votes, article_id]
+    )
+    .then((article) => {
```
Comment (popatre) [inline models/app.model.js]: it's a minor thing, but best practice is not to mix async/await with .then/promise syntax. 
I'd pick one of the other when completing a function

**Review by popatre — CHANGES_REQUESTED** [review]
Just some changes to make please/test to add.

**Review by xViolaine — APPROVED** [review]

**Pull request merged**